### PR TITLE
Lora readme

### PR DESCRIPTION
**Pull request by martinsbruveris**
Adding a README.

### CODE REVIEW

File: tfimm/architectures/lora/README.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+```python
+lora_model = convert_to_lora_model(
+    model, lora_rank=2, lora_train_bias="lora_only"
+)
+```
+
+## Sequential and functional models
+
+The current implementations focusses on models created by subclassing, which is the case
```
Comment (kevin-keraudren) [inline tfimm/architectures/lora/README.md:116]: `implementation` with no `s`

File: tfimm/architectures/lora/README.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+[here](https://github.com/microsoft/LoRA/tree/main). The differences between TensorFlow
+and PyTorch mean that we cannot simply replicate the PyTorch implementation.
+
+- In PyTorch it is possible to mark only LoRA parameters as trainable via
+  ```python
+  # This sets requires_grad to False for all parameters without the 
+  # string "lora_" in their names
+  lora.mark_only_lora_as_trainable(model)
+  ```
+  In TensorFlow it is _not_ possible to change the `trainable` attribute of a variable
```
Comment (kevin-keraudren) [inline tfimm/architectures/lora/README.md:155]: Is it worth mentioning this issue here: https://github.com/tensorflow/tensorflow/issues/47597 ?

File: tfimm/architectures/lora/README.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+- PyTorch allows easy saving and loading a partial state of the model via `state_dict`,
+  ```python
+  torch.save(lora.lora_state_dict(model), "ckpt_lora.pt")
+  model.load_state_dict(torch.load("ckpt_lora.pt"), strict=False)
+  ```
+  In TensorFlow `model.save_weights()` does not allow saving or loading partial states.
+  Something similar can most likely be achieved using TensorFlow checkpoints, but we
+  haven't investigated it yet. In particular, currently we first have to load the
+  regular model into memory and then transfer its weights to the LoRA version.
+
```
Comment (kevin-keraudren) [inline tfimm/architectures/lora/README.md:169]: could we add a list of currently implemented LoRA layers and LoRA models?

File: tfimm/architectures/lora/README.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
+# LoRA: Low-Rank Adaptation for vision models
+
+This module contains TensorFLow code for LoRA layers that can be used for 
+parameter-efficient adaptation of large models. We also integrate `tfimm` models
+with LoRA. For a details on LoRA see the paper
```
[the remark this answers is not in the document]
Comment (martinsbruveris) [reply tfimm/architectures/lora/README.md:5]: Done.

File: tfimm/architectures/lora/README.md (hunk 1 above)
Comment (martinsbruveris) [reply tfimm/architectures/lora/README.md:116]: Thx

File: tfimm/architectures/lora/README.md (hunk 2 above)
Comment (martinsbruveris) [reply tfimm/architectures/lora/README.md:155]: Yes, thx.

File: tfimm/architectures/lora/README.md (hunk 3 above)
Comment (martinsbruveris) [reply tfimm/architectures/lora/README.md:169]: Yes, added that.

**Pull request merged**